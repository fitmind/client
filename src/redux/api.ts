import { CustomerLoginResponse } from '../interfaces/responses/customer-login-response';

import {
    CustomerProfileUpdateActionInterface,
    CustomerSignUpActionInterface,
    ExpertLoginActionInterface,
    UserLoginActionInterface,
    ExpertSignUpActionInterface,
} from './actions/server/server.actions';
import { ExpertLoginResponse } from '../interfaces/responses/expert-login-response';

const API_URL = process.env.REACT_APP_SERVER_URL;

const mode = 'cors';
const credentials = 'include';

export async function apiGetUserDashboard() {
    const res = await fetch(`${API_URL}/user/dashboard`, {
        method: 'get',
        mode,
        credentials,
        headers: { Accept: 'application/json' },
    });
    return res;
}

export async function apiLoginCustomerUser({
    email,
    password,
}: UserLoginActionInterface): Promise<CustomerLoginResponse> {
    const response = await fetch(`${API_URL}/user/login`, {
        method: 'POST',
        headers: { Accept: 'application/json', 'Content-Type': 'application/json' },
        mode,
        credentials,
        body: JSON.stringify({ email, password }),
    });
    return await response.json();
}

export async function apiGetUserMe() {
    const response = await fetch(`${API_URL}/user/me`, {
        method: 'GET',
        headers: { Accept: 'application/json', 'Content-Type': 'application/json' },
        mode,
        credentials,
    });
    return response;
}

export async function apiLogoutCustomerUser() {
    const res = await fetch(`${API_URL}/user/logout`, {
        method: 'post',
        mode,
        credentials,
        headers: { Accept: 'application/json', 'Content-Type': 'application/json' },
    });
    return await res.json();
}

export async function apiSignUpCustomerUser(action: CustomerSignUpActionInterface) {
    const res = await fetch(`${API_URL}/user/register`, {
        method: 'post',
        headers: { Accept: 'application/json', 'Content-Type': 'application/json' },
        body: JSON.stringify({
            email: action.email,
            firstName: action.firstName,
            lastName: action.lastName,
            password: action.password,
            interestedInExpertiseAreas: action.interestedInExpertiseAreas,
            description: action.description,
            phone: action.phone,
        }),
    });
    return res.json();
}

export async function apiSignUpExpertUser(action: ExpertSignUpActionInterface) {
    const res = await fetch(`${API_URL}/expert/register`, {
        method: 'post',
        headers: { Accept: 'application/json', 'Content-Type': 'application/json' },
        body: JSON.stringify({
            email: action.email,
            firstName: action.firstName,
            lastName: action.lastName,
            password: action.password,
            isAnExpertIn: action.isAnExpertIn,
            description: action.description,
            phone: action.phone,
            weeklyAvailability: action.weeklyAvailability,
        }),
    });
    return res.json();
}

export async function apiProfileUpdateCustomerUser(action: CustomerProfileUpdateActionInterface) {
    const res = await fetch(`${API_URL}/user/me`, {
        method: 'put',
        mode,
        credentials,
        headers: { Accept: 'application/json', 'Content-Type': 'application/json' },
        body: JSON.stringify({
            firstName: action.firstName,
            lastName: action.lastName,
            interestedInExpertiseAreas: action.interestedInExpertiseAreas,
            description: action.description,
            phone: action.phone,
        }),
    });
    return res;
}

export async function apiLoginExpertUser({
    email,
    password,
}: ExpertLoginActionInterface): Promise<ExpertLoginResponse> {
    const response = await fetch(`${API_URL}/expert/login`, {
        method: 'POST',
        headers: { Accept: 'application/json', 'Content-Type': 'application/json' },
        mode,
        credentials,
        body: JSON.stringify({ email, password }),
    });
    return await response.json();
}

export async function apiGetExpertMe() {
    const response = await fetch(`${API_URL}/expert/me`, {
        method: 'GET',
        headers: { Accept: 'application/json', 'Content-Type': 'application/json' },
        mode,
        credentials,
    });
    return response;
}

export async function apiLogoutExpertUser() {
    const res = await fetch(`${API_URL}/expert/logout`, {
        method: 'post',
        mode,
        credentials,
        headers: { Accept: 'application/json', 'Content-Type': 'application/json' },
    });
    return await res.json();
}
