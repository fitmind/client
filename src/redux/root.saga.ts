import { all } from 'redux-saga/effects';
import { watchLogoutUserSaga } from './flows/customer-logout/customer-logout.saga';
import { watchSignUpUserSaga } from '../ecosystems/customer-user/customer-register/customer-register.saga';
import { watchLoginUserSaga } from '../ecosystems/customer-user/customer-login/customer-login.saga';
import { watchFetchCustomerUser } from '../ecosystems/customer-user/customer-dashboard/customer-dashboard.saga';
import { watchExpertLoginSaga } from '../ecosystems/expert-user/expert-login/expert-login.saga';
import { watchRegisterExpertSaga } from '../ecosystems/expert-user/expert-register/expert-register.saga';

export function* rootSaga() {
    yield all([
        // user
        watchLoginUserSaga(),
        watchLogoutUserSaga(),
        watchSignUpUserSaga(),
        watchFetchCustomerUser(),

        // expert
        watchExpertLoginSaga(),
        watchRegisterExpertSaga(),
    ]);
}
