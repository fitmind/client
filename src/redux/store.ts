import { Store, createStore, applyMiddleware } from 'redux';
import { ApplicationState, storeInitialState, createRootReducer } from './reducers/root.reducer';
import { composeWithDevTools } from 'redux-devtools-extension';
import createSagaMiddleware from 'redux-saga';
import { rootSaga } from './saga/root.saga';
import { History, createBrowserHistory } from 'history';
import { routerMiddleware } from 'connected-react-router';

// We use hash history because this example is going to be hosted statically.
// Normally you would use browser history.
export const history: History = createBrowserHistory();

export const configureStore = (initialState: ApplicationState): Store<ApplicationState> => {
    const sagaMiddleware = createSagaMiddleware();
    const composeEnhancers = composeWithDevTools({});
    let store = createStore(
        createRootReducer(history),
        initialState,
        composeEnhancers(applyMiddleware(routerMiddleware(history), sagaMiddleware)),
    );

    sagaMiddleware.run(rootSaga);
    return store;
};

const store: Store<ApplicationState> = configureStore(storeInitialState);

export default store;
