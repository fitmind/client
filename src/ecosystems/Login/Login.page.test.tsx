import React from 'react';
import { shallow } from 'enzyme';
import LoginPageWithRouter from './Login.page';
import renderer from 'react-test-renderer';
import { Router } from 'react-router';
import { history } from '../../redux/store';

describe('LoginPage', () => {
    it('renders without crashing', () => {
        shallow(
            <Router history={history}>
                <LoginPageWithRouter />
            </Router>,
        );
    });

    it('should match the snapshot', () => {
        const component = renderer.create(
            <Router history={history}>
                <LoginPageWithRouter />
            </Router>,
        );
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});
