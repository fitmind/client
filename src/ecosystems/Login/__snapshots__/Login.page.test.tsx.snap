// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginPage should match the snapshot 1`] = `
<div
  className="container-fluid"
>
  <div
    className="row"
  >
    <div
      className="col-md-4"
    />
    <div
      className="col"
    >
      <div
        className="sc-bdVaJa fOsasd"
      >
        <div
          className="card border-secondary"
        >
          <h5
            className="card-header"
          >
            Login
          </h5>
          <div
            className="card-body"
          >
            <p
              className="card-text"
            >
              Please enter with your admin login credentials
            </p>
            <form
              className=""
              onSubmit={[Function]}
            >
              <div
                className="form-group"
              >
                <label
                  className="form-label"
                  htmlFor="formBasicEmail"
                >
                  Email address
                </label>
                <input
                  className="form-control"
                  id="formBasicEmail"
                  placeholder="Enter email"
                  type="email"
                />
              </div>
              <div
                className="form-group"
              >
                <label
                  className="form-label"
                  htmlFor="formBasicPassword"
                >
                  Password
                </label>
                <input
                  className="form-control"
                  id="formBasicPassword"
                  placeholder="Password"
                  type="password"
                />
                <small
                  className="text-muted form-text"
                >
                  Password needs to have numbers and letters
                </small>
              </div>
              <button
                className="btn btn-outline-primary btn-block"
                disabled={false}
                type="submit"
              >
                Login
              </button>
            </form>
          </div>
          <div
            className="text-muted card-footer"
          >
            <span
              className="sc-bwzfXH ivdGIV"
              onClick={[Function]}
            >
              To Sign Up click here
            </span>
          </div>
        </div>
      </div>
    </div>
    <div
      className="col-md-4"
    />
  </div>
</div>
`;
